samples/net/sockets/packet/src/packet.c

```diff
@@ -186,6 +187,12 @@ static int send_packet_socket(struct packet_data *packet)
 			}
 		}
 
+		/* If we have received any data, flush it here in order to
+		 * not to leak memory in IP stack.
+		 */
+		(void)recv(packet->send_sock, recv_buffer, sizeof(recv_buffer),
+			   MSG_DONTWAIT);
+
```

Comment:
Maybe it would be better to even recv() in a loop? In theory we might miss some packet (for some reason it would stuck longer in networking stack) and in the next run it would be good to drain as long as we get valid packets:
```
	do {
		ret = recv(packet->send_sock, recv_buffer, sizeof(recv_buffer),
			   MSG_DONTWAIT);
	while (ret == -EAGAIN);
```

Reply:
Sure, that is good idea.